### ISSUE TITLE
rig structure references change to ptr

### ISSUE THREAD

**Issue by mdblack98** [Hamlib/Hamlib#487]
In 5.0 will be changing all internal structure references in rig_caps and such to use pointers.
Problem with shared libaries was created when hamlib_port_t had a variable added (i.e. flushx) which changed offsets of everything past hamlib_port_t in rig_caps.
rig_caps will be changes to make any such items pointers instead of static structures.

**Comment by N0NB** [comment]
@GeoBaltz I think you're working toward completing this in 5.0.

**Comment by GeoBaltz** [comment]
Yes, Nate, I am.  Even with all the prep work done over the last year and a half I'm still not sure what I'll run into at the end.  I do intend to have the application side ready for 4.7, so the user/app side source should work on either 4.7 or 5.0.

Aside from the obvious benefit of not having structures within structures, I'd like to make rig.h enough to call the API, with as little as possible of the internals of Hamlib in it.  This sort of future-proofs the dll problems of the past.

It might be good to add a reference to this issue to the 5.0 feature list.

**Comment by N0NB** [comment]
Good idea.  I will do so.
